component/xload/blockpool.go

```diff
@@ -142,8 +148,18 @@ func (pool *BlockPool) GetBlock(priority bool) *Block {
 
 // TryGet a block from the pool. If the pool is empty, wait till a block is released back to the pool
 func (pool *BlockPool) tryGet() *Block {
+	var block *Block = nil
+
+	select {
+	case <-pool.ctx.Done():
+		err := fmt.Errorf("Failed to Allocate Buffer as the process was cancelled, Len (blocksCh: %d, priorityCh: %d), MaxBlocks: %d",
+			len(pool.blocksCh), len(pool.priorityCh), pool.maxBlocks)
+		log.Warn("BlockPool::GetBlock : %v", err)
```

Comment:
This should be error log

Reply:
error that we return will be logged by the caller anyway, so we can actually skip this error log here I beleive.